Add optional time residuals MAD filter for split vertex rejection + related

shahor02 opened the pull request.

shahor02 committed:
Fixes for TrackStudy, add different t,z statistics.

shahor02 committed:
Add MAD estimator to MathUtils fit collection

alibuild commented:
Error while checking build/O2/fullCI for 245f8cc2b5d5bf7b14c403cf67ef42994bd6bd0c at 2023-10-09 22:56:
```
No log files found
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12043/245f8cc2b5d5bf7b14c403cf67ef42994bd6bd0c/build_O2_fullCI/pretty.html).

shahor02 committed:
Optional policies for matching timestamp outliers

If ITS-TPC match timestamp is outside of the relevant ITS ROF time bracket, select behavious from
tpcitsMatch.TimeOutliersPolicy options:
Tolerate: take as is
Adjust  : adjust to closest ITS ROF bracket boundary
Reject  : reject match
Default is Adjust

shahor02 committed:
Add getter for squared time error

shahor02 committed:
SVertexer reports number of DCAfitter calls

shahor02 committed:
Fixes for InteractionTag, TrackStudy, add different t,z statistics

shahor02 committed:
Extra cleanup with t,z MAD estimators
